src/Symfony/Component/DependencyInjection/Definition.php

```diff
@@ -830,6 +835,55 @@ public function isAbstract()
     }
 
     /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @param bool   $boolean
+     * @param string $template Template message to use if the definition is deprecated
+     *
+     * @return Definition the current instance
+     *
+     * @api
+     */
+    public function setDeprecated($boolean = true, $template = null)
+    {
+        $this->deprecation['status'] = (bool) $boolean;
+
+        if (null !== $template) {
+            $this->deprecation['template'] = $template;
+        }
+
+        return $this;
+    }
+
+    /**
+     * Whether this definition is deprecated, that means it should not be called
+     * anymore.
+     *
+     * @return bool
+     *
+     * @api
+     */
+    public function isDeprecated()
+    {
+        return $this->deprecation['status'];
+    }
+
+    /**
+     * Message to use if this definition is deprecated.
+     *
+     * @param string $id Service id relying on this definition
+     *
+     * @return string
+     *
+     * @api
+     */
+    public function getDeprecationMessage($id)
```

Comment:
I used `%service_id% in the message template then, but I left `$id` as a parameter here, as this is the definition (Service id), this should be explicit enough, as this is what is used in the other places